hydromt/data_adapter/rasterdataset.py

```diff
@@ -597,10 +597,17 @@ def _slice_spatial_dimensions(
             bbox_str = ", ".join([f"{c:.3f}" for c in bbox])
             logger.debug(f"Clip to [{bbox_str}] (epsg:{epsg}))")
             ds = ds.raster.clip_bbox(bbox, buffer=buffer, align=align)
+
             # if np.any(np.array(ds.raster.shape) < 2):
             # check if bbox is fully covered
+            def _lt_or_close(a: float, b: float) -> bool:
+                return np.isclose(a, b) or a < 0
```

Comment:
Shouldn't this be `a < b` instead of `a < 0`

```suggestion
                return np.isclose(a, b) or a < b
```

Alternatively this check could also be done before the clip_bbox (L599) to circumvent most rounding errors. But probably not all for some edge cases, so the new check is still better.